File: AlgoExpert/cpp/generate_document.cc
/* Notes
Difficulty: Easy
*/

#include <string>
#include <unordered_map>

bool generateDocument(std::string characters, std::string document) {
  // Write your code here.
  std::unordered_map<char, int> char_count;
  std::unordered_map<char, int> doc_count;
  for (const char ch : characters) char_count[ch]++;
  for (const char ch : document) doc_count[ch]++;

  for (const auto &[key, value] : doc_count) {
    if (char_count[key] < value) return false;
  }
  return true;
}

File: AlgoExpert/python/generate_document.py
"""
Difficulty: Easy
"""

from collections import Counter


def generateDocument(characters: str, document: str) -> bool:
    # Write your code here.
    chars_count = Counter(characters)
    document_count = Counter(document)

    for key, value in document_count.items():
        if key not in chars_count or chars_count[key] < value:
            return False
    return True
